packages/client/src/clients/in-app-messages/message-client.ts

```diff
@@ -265,4 +261,36 @@ export class InAppMessagesClient {
     const items = Array.isArray(itemOrItems) ? itemOrItems : [itemOrItems];
     return items.map((item) => item.id);
   }
+
+  private getMessageVisibility(
```

Comment:
Open to feedback on this method. Tried a few things and landed here. Some alternatives:
- Just manually writing checks for each status (`if engagement_status.includes('seen') && !message.seen_at return false`. Was pretty verbose
- Doing it fully dynamically like by checking `message[status]` but that was a bit harder to read and understand